iothub/device/src/InternalClient.cs

```diff
@@ -1974,11 +1974,17 @@ internal void ValidateModuleTransportHandler(string apiName)
         public void Dispose()
         {
             InnerHandler?.Dispose();
+            InnerHandler = null;
             _methodsDictionarySemaphore?.Dispose();
             _moduleReceiveMessageSemaphore?.Dispose();
             _fileUploadHttpTransportHandler?.Dispose();
             _deviceReceiveMessageSemaphore?.Dispose();
             _twinDesiredPropertySemaphore?.Dispose();
+#if !NET451
+            Certificate?.Dispose();
+            Certificate = null;
+#endif
+            IotHubConnectionString?.TokenRefresher?.Dispose();
```

Comment:
I decided to dispose whatever is needed here. Making the IotHubConnectionString class did not seem right as we really do not want that dispose to be called anywhere else.